utils/googleutils/googleutils.go

```diff
@@ -3,12 +3,55 @@ package googleutils
 import (
 	"fmt"
 
+	"github.com/rudderlabs/rudder-server/config"
 	"golang.org/x/oauth2/google"
 )
 
+const (
+	EMPTY_CREDS   = "{}"
+	WI_CONFIG_KEY = "workloadIdentity"
+)
+
 func CompatibleGoogleCredentialsJSON(jsonKey []byte) error {
 	if _, err := google.ConfigFromJSON(jsonKey); err == nil {
 		return fmt.Errorf("Google Developers Console client_credentials.json file is not supported")
 	}
 	return nil
 }
+
+func ShouldSkipCredentialsInit(credentials string) bool {
+	return IsGKEEnabledWorkload() && IsCredentialsStringEmpty(credentials)
+}
+
+/*
+	The credentials are deemed to be empty when either the field credentials is
+	sent as empty string or when the field is set with "{}"
+
+	Note: This is true only for workload identity enabled rudderstack data-plane deployments
+*/
+func IsCredentialsStringEmpty(credentials string) bool {
+	return (credentials == "" || credentials == EMPTY_CREDS)
+}
+
+/*
+	We would check for rudder-server configuration for workload identity for google destinations
+*/
+func IsGKEEnabledWorkload() bool {
+	workloadType := config.GetString(fmt.Sprintf("%s.type", WI_CONFIG_KEY), "")
+	if workloadType == "GKE" {
+		/*
+			The workload when deployed in Google GKE cluster, this is to be used
+
+			workloadIdentity.projectId -> The project which contains the workload identity pool in GKE
+			workloadIdentity.gsaName -> The google IAM Service account name
+			workloadIdentity.gsaProjectId -> The project in which the mentioned google IAM service account is present
+
+			Note: We are currently only validating the schema rather than the values
+		*/
+		workloadProjectId := config.GetString(fmt.Sprintf("%s.gwiProjectId", WI_CONFIG_KEY), "")
+		workloadGsaName := config.GetString(fmt.Sprintf("%s.gsaName", WI_CONFIG_KEY), "")
+		workloadGsaProjectId := config.GetString(fmt.Sprintf("%s.gsaProjectId", WI_CONFIG_KEY), "")
+		return workloadProjectId != "" && workloadGsaName != "" && workloadGsaProjectId != ""
```

Comment:
Why validate these? I'd just check the workload's enabled status (missing) and its type.

Reply:
> check the workload's enabled status (missing) and its type.

Workload enabled is not present in the configuration recommended by you right ?! Also it didn't make sense to be available there.
The additional configuration is to make sure they supply all these values in the configuration as they'd be required for the deployment

Reply:
I meant from `serverConfigOverride`.
```
serverConfigOverride:
  workloadIdentityEnabled: true
workloadIdentity:
  type: GKE # EKS for Amazon support
  ...
```

I disagree that the additional parameters should be validated here. The service consuming them should bother about them.

Reply:
These values are being used while creating the rudderstack deployment. Any insight into this would be good
It was provided as a suggestion by you to keep it enabled by default right ?! 
Why is `workloadIdentityEnabled` is being put in `serverConfigOverride` ?

Reply:
Removed the whole thing, kindly check and let me know @saikumarrs